homeassistant/components/eufylife_ble/sensor.py

```diff
@@ -0,0 +1,211 @@
+    _attr_native_unit_of_measurement = UnitOfMass.KILOGRAMS
+    _attr_device_class = SensorDeviceClass.WEIGHT
+
+    def __init__(self, data: EufyLifeData) -> None:
+        """Initialize the real-time weight sensor entity."""
+        super().__init__(data)
+        self._attr_unique_id = f"{data.address}_real_time_weight"
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the native value."""
+        if self._data.client.state is not None:
+            return self._data.client.state.weight_kg
+        return None
+
+    @property
+    def suggested_unit_of_measurement(self) -> str | None:
+        """Set the suggested unit based on the unit system."""
+        if self.hass.config.units is US_CUSTOMARY_SYSTEM:
+            return UnitOfMass.POUNDS
+
+        return UnitOfMass.KILOGRAMS
+
+
+class EufyLifeWeightSensorEntity(RestoreEntity, EufyLifeSensorEntity):
+    """Representation of an EufyLife weight sensor."""
+
+    _attr_name = "Weight"
+    _attr_native_unit_of_measurement = UnitOfMass.KILOGRAMS
+    _attr_device_class = SensorDeviceClass.WEIGHT
+
+    _weight_kg: float | None = None
+
+    def __init__(self, data: EufyLifeData) -> None:
+        """Initialize the weight sensor entity."""
+        super().__init__(data)
+        self._attr_unique_id = f"{data.address}_weight"
+
+    @property
+    def available(self) -> bool:
+        """Determine if the entity is available."""
+        return True
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the native value."""
+        return self._weight_kg
+
+    @property
+    def suggested_unit_of_measurement(self) -> str | None:
+        """Set the suggested unit based on the unit system."""
+        if self.hass.config.units is US_CUSTOMARY_SYSTEM:
+            return UnitOfMass.POUNDS
+
+        return UnitOfMass.KILOGRAMS
+
+    @callback
+    def _handle_state_update(self, *args: Any) -> None:
+        """Handle state update."""
+        state = self._data.client.state
+        if state is not None and state.is_final:
+            self._weight_kg = state.weight_kg
+
+        super()._handle_state_update(args)
+
+    async def async_added_to_hass(self) -> None:
+        """Restore state on startup."""
+        await super().async_added_to_hass()
+
+        last_state = await self.async_get_last_state()
+        if not last_state or last_state.state == STATE_UNAVAILABLE:
+            return
+
+        with suppress(ValueError):
+            last_weight = float(last_state.state)
+            last_weight_unit = last_state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+
+            # Since the RestoreEntity stores the state using the displayed unit,
+            # not the native unit, we need to convert the state back to the native
+            # unit.
+            self._weight_kg = MassConverter.convert(
+                last_weight, last_weight_unit, self.native_unit_of_measurement
+            )
```

Comment:
It seems like it would be a bug if the state wasn't a float

Reply:
See my reply above about the `int` one.

Reply:
Removed suppression of the `ValueError`.